src/test/java/com/thebois/models/beings/BeingTests.java

```diff
@@ -160,4 +183,79 @@ public void equalsReturnsFalseForNotEqualBeings(final AbstractBeing first,
         assertThat(first).isNotEqualTo(second);
     }
 
+    @Test
+    public void pawnBuildsHouses() {
+        // Arrange
+
+        final IFinder mockFinder = Mockito.mock(IFinder.class);
+        final Position position = new Position();
+        final IStructure structure = new House(position);
+        when(mockFinder.findNearestStructure(any())).thenReturn(Optional.of(structure));
+        final Pawn pawn = new Pawn(position, position, Mockito.mock(Random.class), mockFinder);
+
+        // Act
+        final float initialBuildStatus = structure.builtStatus();
+        pawn.update();
+        final float finalBuildStatus = structure.builtStatus();
+
+        // Assert
+
+        assertThat(initialBuildStatus).isLessThan(finalBuildStatus);
+    }
+
+    @Test
+    public void pawnDoNotBuildsHouses() {
+        // Arrange
+        final IFinder mockFinder = Mockito.mock(IFinder.class);
+        final Position position = new Position(0, 0);
+        final Position farAway = new Position(50, 50);
+        final IStructure structure = new House(farAway);
+        when(mockFinder.findNearestStructure(any())).thenReturn(Optional.of(structure));
+        final Pawn pawn = new Pawn(position, position, Mockito.mock(Random.class), mockFinder);
+
+        // Act
+        final float initialBuildStatus = structure.builtStatus();
+        pawn.update();
+        final float finalBuildStatus = structure.builtStatus();
+
+        // Assert
+        // Value did not change because Pawn was not close enough
+        assertThat(initialBuildStatus).isEqualTo(finalBuildStatus);
+    }
+
+    @Test
+    public void doesPawnWalkToNearestUnBuiltBuilding() {
+        // Arrange
+        final Position positionA = new Position(1, 0);
+        final Position positionB = new Position(2, 0);
+        final Position positionC = new Position(3, 0);
+
+        final IStructure unBuiltStructure = Mockito.mock(IStructure.class);
+        final IStructure builtStructure = Mockito.mock(IStructure.class);
+
+        when(unBuiltStructure.getPosition()).thenReturn(positionA);
+        when(builtStructure.getPosition()).thenReturn(positionC);
+        when(unBuiltStructure.builtStatus()).thenReturn(0f);
+        when(builtStructure.builtStatus()).thenReturn(1f);
+
+        final Collection<Optional<IStructure>> structures = new ArrayList<>();
+        structures.add(Optional.of(builtStructure));
+        structures.add(Optional.of(unBuiltStructure));
+
+        final IFinder mockFinder = Mockito.mock(IFinder.class);
+
+        when(mockFinder.findNearestStructures(any(), anyInt())).thenReturn(structures);
+        // builtStructure would be found first by the current finder
+        when(mockFinder.findNearestStructure(any())).thenReturn(Optional.of(builtStructure));
```

Comment:
question

Does `Pawn` use both `findNearestStructures` and `findNearestStructure`, and if so, why use both?